include/GafferScene/Shader.h

```diff
@@ -142,6 +142,9 @@ class Shader : public Gaffer::DependencyNode
 		virtual void attributesHash( const Gaffer::Plug *output, IECore::MurmurHash &h) const;
 		virtual IECore::ConstCompoundObjectPtr attributes( const Gaffer::Plug *output ) const;
 
+		Gaffer::StringPlug *attributeSuffixPlug();
```

Comment:
I think we're intending the user to be able to modify the values of this plug directly aren't we? If that's the case then I think it should be public, up there just before the parameters plug.